libraries/botbuilder-azure/src/cosmosDbStorage.ts

```diff
@@ -128,6 +133,15 @@ export class CosmosDbStorage implements Storage {
         }
 
         this.client = new DocumentClient(settings.serviceEndpoint, { masterKey: settings.authKey }, policy);
+
+        // Note: hack, however it works with our version
+        if (settings.agent) {
```

Comment:
Is there an issue tracking the fix for the agent and/or moving away from this hack in the future?

Reply:
The `documentdb` package is [already deprecated](https://www.npmjs.com/package/documentdb), so we won't ever upgrade it. As I'm sure we have many legacy customers using this package, it seems reasonable to take this.